src/Data/FormURLEncoded.purs

```diff
@@ -35,3 +37,12 @@ encode = String.joinWith "&" <<< map encodePart <<< toArray
     encodePart = case _ of
       Tuple k Nothing -> unsafeEncodeURIComponent k
       Tuple k (Just v) -> unsafeEncodeURIComponent k <> "=" <> unsafeEncodeURIComponent v
+ 
+-- | Decode `FormURLEncoded` from `application/x-www-form-urlencoded`.
+decode :: String -> FormURLEncoded
+decode = FormURLEncoded <<< catMaybes <<< map decodePart <<< String.split (Pattern "&")
+  where
+    decodePart = String.split (Pattern "=") >>> case _ of
+      [k, v] -> Just $ Tuple (unsafeDecodeURIComponent k) (Just $ unsafeDecodeURIComponent v)
+      [k]    -> Just $ Tuple (unsafeDecodeURIComponent k) Nothing
+      _      -> Nothing
```

Comment:
I'm not particularly familiar with `decodeURIComponent`, but is it possible that a key or value could cause an exception? For example, the MDN documentation provides this pair of examples, the first of which ought to succeed and the second of which ought to fail:

```js
decodeURIComponent('JavaScript_%D1%88%D0%B5%D0%BB%D0%BB%D1%8B');
// "JavaScript_шеллы"

try { 
  var a = decodeURIComponent('%E0%A4%A'); 
} catch(e) { 
  console.error(e); 
}

// URIError: malformed URI sequence
```

neither of which contains the explicit pattern `=` or `&`, and I'm not sure which part of it causes the exception thrown in the second case.

Do you mind clarifying how the `decode` function avoids exceptions, and handles the same cases that the JavaScript `decodeURIComponent` function does? (Feel free to correct me if this question doesn't seem pertinent).

Reply:
Ideally the `decode` function in this library would:

1. Successfully decode the same cases that the `decodeURIComponent()` function does
2. Return `Nothing` for cases that would cause an exception in `decodeURIComponent()`

Reply:
Now that there are safe versions in `-globals`, can we update this @nsaunders?